make_test_images/src/compare_manifests.rs

```diff
@@ -149,129 +146,258 @@ fn gather_manifests(manifest_store: &Reader, manifest_label: &str, labels: &mut
-                            == cap2.get(3).map_or("", |m| m.as_str())
-                    {
-                        return;
+        (Value::Array(arr1), Value::Array(arr2)) => {
+            // Use custom identifier logic for validation_results arrays
+            let use_custom_id = !arr1.is_empty()
+                && arr1.iter().all(|v| v.is_object())
+                && arr1.iter().all(|v| get_array_identifier(path, v).is_some());
+            // For ingredients and actions arrays, use simplified matching
+            if path.contains("ingredients") || path.contains("actions") {
+                if arr1.len() != arr2.len() {
+                    issues.push(format!(
+                        "Array length changed at {path}: {} vs {}",
+                        arr1.len(),
+                        arr2.len()
+                    ));
+                } else {
+                    // Match by title for ingredients
+                    if path.contains("ingredients") {
+                        let mut map1 = std::collections::HashMap::new();
+                        let mut map2 = std::collections::HashMap::new();
+                        for v in arr1 {
+                            if let Some(title) = v.get("title").and_then(|t| t.as_str()) {
+                                map1.insert(title, v);
+                            }
+                        }
+                        for v in arr2 {
+                            if let Some(title) = v.get("title").and_then(|t| t.as_str()) {
+                                map2.insert(title, v);
+                            }
+                        }
+                        for title in map1.keys() {
+                            if let (Some(v1), Some(v2)) = (map1.get(title), map2.get(title)) {
+                                compare_json_values(&format!("{path}[{title}]"), v1, v2, issues);
+                            }
+                        }
+                    } else if path.contains("actions") {
+                        let mut map1 = std::collections::HashMap::new();
+                        let mut map2 = std::collections::HashMap::new();
+                        for v in arr1 {
+                            if let Some(action) = v.get("action").and_then(|a| a.as_str()) {
+                                map1.insert(action, v);
+                            }
+                        }
+                        for v in arr2 {
+                            if let Some(action) = v.get("action").and_then(|a| a.as_str()) {
+                                map2.insert(action, v);
+                            }
+                        }
+                        for action in map1.keys() {
+                            if let (Some(v1), Some(v2)) = (map1.get(action), map2.get(action)) {
+                                compare_json_values(&format!("{path}[{action}]"), v1, v2, issues);
+                            }
+                        }
```

Comment:
This is fine but it would be nice if it was separated out into its own function.